### PR TITLE
chore: Updating pre-commit hook for staged files to not run spotless check if its a merge commit

### PR DESCRIPTION
**Pull request by ankitakinger**
## Description

Updating pre-commit hook for staged files to not run spotless check if its a merge commit

#### PR fixes following issue(s)
Fixes # (issue number)

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/.husky/check-staged-files.sh [hunk 1]
```diff
@@ -3,7 +3,9 @@
 is_server_change=$(git diff --cached --name-only | grep -c "app/server")
 is_client_change=$(git diff --cached --name-only | grep -c "app/client")
 
-if [ "$is_server_change" -ge 1 ]; then
+is_merge_commit=$(git rev-parse -q --verify MERGE_HEAD)
+
+if [[ "$is_server_change" -ge 1 && ! "$is_merge_commit" ]]; then
```
Comment (KelvinOm) [inline app/client/.husky/check-staged-files.sh:8]: This also applies to FE checks. I think it's better to write code like this:

```
if [ "$is_merge_commit" ]; then
  echo "Skipping server and client checks for merge commit"
else
  if [ "$is_server_change" -ge 1 ]; then
      ...
  fi

  if [ "$is_client_change" -ge 1  ]; then
    ...
  fi
fi
```

File: app/client/.husky/check-staged-files.sh [hunk 2]
```diff
@@ -3,7 +3,9 @@
 is_server_change=$(git diff --cached --name-only | grep -c "app/server")
 is_client_change=$(git diff --cached --name-only | grep -c "app/client")
 
-if [ "$is_server_change" -ge 1 ]; then
+is_merge_commit=$(git rev-parse -q --verify MERGE_HEAD)
```
Comment (KelvinOm) [inline app/client/.husky/check-staged-files.sh:6]: I'm not familiar with this syntax, it would be worth to test locally on different branches.

Comment (ankitakinger) [reply app/client/.husky/check-staged-files.sh:6]: I;m using this syntax on EE. Sure on this one. Wasn't sure of the syntax for the if condition on line 8.

**Review by KelvinOm — DISMISSED** [review]

File: app/client/.husky/check-staged-files.sh [hunk 3]
```diff
@@ -3,22 +3,28 @@
...
 is_client_change=$(git diff --cached --name-only | grep -c "app/client")
 
-if [ "$is_server_change" -ge 1 ]; then
-  echo "Running Spotless check ..."
+is_merge_commit=$(git rev-parse -q --verify MERGE_HEAD)
+
+if [ "$is_merge_commit" ]; then
+  echo "Skipping server and client checks for merge commit"
+else 
+  if [ "$is_server_change" -ge 1 ]; then
```
Comment (aswathkk) [inline app/client/.husky/check-staged-files.sh:11]: The syntax is `elif`. Not `else if`

**Comment by aswathkk** [comment]
@ankitakinger My bad! Sorry for adding the comment early

**Pull request merged**